Refactor Keystore

gemdev111 opened the pull request.

gemdev111 committed:
WalletConnectorSigner removed keystore

gemdev111 committed:
Removed Keystore dependecies from services

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
fix packages

gemdev111 committed:
Merge branch 'main'

gemcoder21 commented on Services/WalletsService/Sources/WalletsService.swift:
```diff
@@ -79,6 +80,10 @@ public struct WalletsService: Sendable {
         self.addressStatusService = addressStatusService
     }
 
+    public var walletsCount: Int {
```
```suggestion
    public var walletsCount: Int {
```

I would recommend to add throwable method here to get the number and view model to handle try?

gemcoder21 commented on Services/WalletsService/Sources/DiscoveryAssetsProcessor.swift:
```diff
@@ -5,28 +5,30 @@ import AssetsService
 
     func discoverAssets(for walletId: WalletId, preferences: WalletPreferences) async throws {
-        let wallet = try keystore.getWallet(walletId)
+        guard let wallet = try walletStore.getWallet(id: walletId.id) else {
```
maybe use `WalletService` here to get the wallet to avoid throwing an error? `WalletService` should have getWallet method already

gemcoder21 commented on Services/WalletsService/Sources/BalanceUpdateService.swift:
```diff
@@ -3,28 +3,32 @@
     // add asset to asset store and create balance store record
     func enableBalances(for walletId: WalletId, assetIds: [AssetId]) throws {
-        let wallet = try keystore.getWallet(walletId)
+        guard let wallet = try walletStore.getWallet(id: walletId.id) else {
```
same here, use `try walletService.getWallet()`

gemcoder21 commented on Services/WalletsService/Sources/BalanceUpdateService.swift:
```diff
@@ -3,28 +3,32 @@
 
     func updateBalance(for walletId: WalletId, assetIds: [AssetId]) async throws {
-        let wallet = try keystore.getWallet(walletId)
+        guard let wallet = try walletStore.getWallet(id: walletId.id) else {
```
same here, use `try walletService.getWallet()`

gemdev111 committed:
Keystore updates
Added WalletSessionService reponsible for handling selecting current wallet, observering
Moved logic of selecting current wallet from Keystore to WalletSessionService
Updated ManageWalletService with WalletSessionService
Updates other services to handle selection of getting wallet via WalletSessionService or ManageWalletService
Added Tests for WalletSessionService
Adjusted tests after updates
Added Sendable, removed main actor from keystore, fixed xcode warnings
Adjusted Tab navigations to work with ManageWalletService, WalletSessionService current wallet observation
Added utility QueryObserverModifier to move grdb observing into view model, integrated in transactions as showcase
Extended ObservablePreferences with CurrentWalletId for reactive observering
Minor refacoring & removed unused dependencies

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
minor update

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Update core

gemcoder21 committed:
Merge branch 'main' into keystore

DRadmir commented on Features/WalletTab/Sources/Scenes/WalletScene.swift:
```diff
@@ -166,16 +172,13 @@ public struct WalletScene: View {
-                }
+        Task {
+            do {
+                try await model.fetch(walletId: model.wallet.walletId, assets: assets)
```
Can we get walletId directly in model?
```suggestion
                try await model.fetch(assets: assets)
```

DRadmir commented on Packages/Keystore/Sources/LocalKeystore.swift:
```diff
@@ -3,99 +3,44 @@ import Primitives
     public var wallets: [Primitives.Wallet] {
         do {
-            return try walletStore.getWallets()
+            return try walletSessionService.getWallets()
```
Throws error not handled, mb better use `try?`

```suggestion
            try? walletSessionService.getWallets() ?? []
```

or

```suggestion
            try? walletSessionService.getWallets().or([])
```

DRadmir commented on Services/WalletSessionService/Sources/Errors/WalletSessionServiceError.swift:
```diff
@@ -0,0 +1,15 @@
+    case noWallet
+    case noWalletId
+
+    var errorDescription: String? {
```
Will delivered this error for to user? Should we use Localized?

gemdev111 replied:
This errors mostly for developers and shouldn't be delivered to users. LocalKeystore needs more refactoring, this pr mostly to remove observering of current wallet from keystore, I think in a future refactoring it makes sense to localize all errors related keystore + wallet session + walletmanager, but not only some part of it

DRadmir commented on Services/WalletSessionService/Tests/WalletSessionServiceTests.swift:
```diff
@@ -0,0 +1,79 @@
+
+    @Test
+    func testCurrentWalletReturnsCorrectWallet() throws {
+        let walletStore = WalletStore.mock()
```
Avoid code repetition by creating a function to add mock wallets

gemdev111 committed:
Merge branch 'main'

#Conflicts:
#	Gem/Root/Navigation/NFT/CollectionsNavigationStack.swift

gemdev111 committed:
WalletScene removed wallet observering
Removed unused .task fetch, onChange handles it.
Simplified tests
Minor updates

gemdev111 committed:
Merge branch 'main'

#Conflicts:
#	Gem/Swap/Navigation/SwapNavigationView.swift

gemdev111 committed:
fix info sheet naming

gemdev111 committed:
back wallet bar dbwallet observer
